### PR TITLE
Detect equal distribution keys better, to avoid unnecessary Motion no…

### PR DESCRIPTION
**Pull request by hlinnaka**
…des.

More fallout from the equivalence classes merge.

### CODE REVIEW

**Comment by atris** [comment]
+1

File: src/backend/cdb/cdbpathlocus.c [hunk 1]
```diff
@@ -40,6 +40,56 @@ cdb_build_distribution_pathkeys(PlannerInfo      *root,
  */
 bool cdbpathlocus_querysegmentcatalogs = false;
 
+/*
+ * Are two pathkeys equal?
+ *
+ * This is the roughly the same as compare_pathkeys, but compare_pathkeys
```
Comment (danielgustafsson) [inline src/backend/cdb/cdbpathlocus.c]: s/is the roughly/is roughly/

**Comment by danielgustafsson** [comment]
:+1: with a small nitpick

**Comment by hlinnaka** [comment]
Fixed and pushed, thanks

**Pull request closed without merging**